src/SepaController.php

```diff
@@ -15,6 +15,22 @@ public function __construct() {
     $this->intentSettings = [
       'payment_method_types' => ['sepa_debit'],
     ];
+    // Stripe requires a name and an email address for SEPA payments,
+    // set default to display those fields if not already on the form.
+    $this->controller_data_defaults['input_settings'] = [
+      'billing_details' => [
+        'name' => [
+          'enabled' => 1,
+          'display' => 'ifnotset',
+          'required' => 1,
+        ],
+        'email' => [
+          'enabled' => 1,
+          'display' => 'ifnotset',
+          'required' => 1,
+        ],
+      ]
+    ];
```

Comment:
These fields should then also have `$element['#required']` in the method configuration.

Reply:
Ok… what is the "method configuration" if that’s not it? :thinking:

Reply:
I meant `CustomerDataForm::fields()`. The element definitions there are used to generate the method configuration form.

Reply:
I don’t really understand? In `CustomerDataForm::configurationForm()`, the `controller_data_defaults` are merged with `CustomerDataForm::fields()`. Why should they be in the elements definitions before that?

Reply:
Because a field being actually required by the payment provider is different from the setting to make it required by choice.